File: 7kyu/Python/FilterUnusedDigits.py
'''
Given a varying number of integer arguments, return the digits that are not present in any of them.

Example:
[12, 34, 56, 78]  =>  "09"
[2015, 8, 26]     =>  "3479"
Note: the digits in the resulting string should be sorted.
'''
# METHOD 1
def unused_digits(*numbers):
    allDigits = '0123456789'
    result = ''
    numbers = str(numbers)
    
    for num in allDigits:
        if num not in numbers:
            result += num
    return result

# METHOD 2
def unused_digits(*numbers):
    # if the number from all digits is not in the string version of the arguments, join and return it
    return "".join(number for number in "0123456789" if number not in str(numbers))

File: 7kyu/JavaScript/FilterUnusedDigits.js
/*
Given a varying number of integer arguments, return the digits that are not present in any of them.

Example:
[12, 34, 56, 78]  =>  "09"
[2015, 8, 26]     =>  "3479"
Note: the digits in the resulting string should be sorted.
*/

// METHOD 1
const unusedDigits = (...a) => {
  // array of all the digits
  const allDigits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
  
  // create array of each individual digit in order
  a = a.join('').split('').sort((a, b) => a - b).map(e => +e);
  
  // filter allDigits to only the elements that are missing in the original array & return
  return allDigits.filter(e => !a.includes(e)).join('')
}

// METHOD 1 but more concise
const unusedDigits = (...a) => {
  const allDigits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
  return allDigits.filter(e => !a.join('').includes(e)).join('')
}

// METHOD 1 most concise
const unusedDigits = (...a) => [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].filter(e => !a.join('').includes(e)).join('')
